src/controllers/controller.line.js

```diff
@@ -209,6 +209,19 @@ module.exports = function(Chart) {
 			};
 		},
 
+		calculatePointX: function(value, index, datasetIndex){
+			var me = this;
+			var meta = me.getMeta();
+			var labels = me.chart.data.labels;
+			var dataset = me.getDataset();
+			var xScale = me.getScaleForId(meta.xAxisID);
+			if (labels && labels.length === 1 || dataset.data.length === 1) {
+				return xScale.getPixelForValue(typeof value === 'object' ? value : NaN, index, datasetIndex, true);
+			} else {
+				return xScale.getPixelForValue(typeof value === 'object' ? value : NaN, index, datasetIndex, me.chart.isCombo);
+			}
+		},
```

Comment:
I really like splitting logic in small functions, but in this case I would not have created `calculatePointX`. Everything is already available in `updateElement`, and since `calculatePointX` is only called from `updateElement` (which one can be call many times), the extra `getMeta()`, `getDataset()`, `getScaleForId()` access are really a waste of performance.  Also, we should avoid code duplication when the only difference is one parameter. So the code in `calculatePointX` could fit in 3 lines in `updateElement`:
```javascript
var labels = me.chart.data.labels || [];
var includeOffset = (labels.length === 1 || dataset.data.length === 1) || me.chart.isCombo;
// ...
x = xScale.getPixelForValue(typeof value === 'object' ? value : NaN, index, datasetIndex, includeOffset);
```

**But** since `calculatePointY` already exists, I'm not sure what to think about `calculatePointX`.  Maybe  `calculatePointX` and `calculatePointY` should have extra parameters: (`xScale`, `dataset`) and (`yScale`) respectively and thus avoid extra lookup time. Actually, I like this solution better. What do you think?